### PR TITLE
fix: [CI-9000]: Add prefix for log service streams

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51522)

### CODE REVIEW

**Comment by smjt-h** [comment]
trigger uts

**Comment by smjt-h** [comment]
trigger ut

File: product/log-service/handler/stream.go [hunk 1]
```diff
@@ -128,9 +129,10 @@ func HandleClose(logStream stream.Stream, store store.Store, scanBatch int64) ht
 				g.Go(func() error {
 					return logStream.CopyTo(ctx, k, bwc)
 				})
-
+				//Remove Stream Prefix
```
Comment (vistaarjuneja) [inline product/log-service/handler/stream.go:132]: why do we need to remove log key here? Couldn't we just have key addition and removal at the base layer and everything else remains the same?

File: product/log-service/stream/redis/redis.go [hunk 2]
```diff
@@ -362,3 +412,17 @@ func (r *Redis) expiryWatcher(expiry time.Duration, scanBatch int64) {
 	}
 	logrus.Infof("done running expiry watcher thread in %s time and expired %d keys", time.Since(st), cnt)
 }
+
+func CreateLogStreamPrefixedKey(key string) string {
```
Comment (vistaarjuneja) [inline product/log-service/stream/redis/redis.go:416]: you can keep these as internal functions so we only use them at the base layer

File: product/log-service/stream/redis/redis.go [hunk 3]
```diff
@@ -362,3 +412,17 @@ func (r *Redis) expiryWatcher(expiry time.Duration, scanBatch int64) {
...
+	if !strings.HasPrefix(key, stream.LogStreamPrefix) {
```
Comment (vistaarjuneja) [inline product/log-service/stream/redis/redis.go:417]: remove the check for has prefix - we should just add it

File: product/log-service/handler/stream.go [hunk 4]
```diff
@@ -128,9 +129,10 @@ func HandleClose(logStream stream.Stream, store store.Store, scanBatch int64) ht
...
+				blobKey := redis.RemoveLogStreamPrefixedKey(key)
 				g.Go(func() error {
-					return store.Upload(ctx, k, br)
+					return store.Upload(ctx, blobKey, br)
```
Comment (vistaarjuneja) [inline product/log-service/handler/stream.go:135]: From API perspective nothing changes - we should still receive the same key

File: product/log-service/handler/stream.go (hunk 1 above)
Comment (smjt-h) [reply product/log-service/handler/stream.go:132]: done

File: product/log-service/stream/redis/redis.go (hunk 2 above)
Comment (smjt-h) [reply product/log-service/stream/redis/redis.go:416]: done

File: product/log-service/stream/redis/redis.go (hunk 3 above)
Comment (smjt-h) [reply product/log-service/stream/redis/redis.go:417]: done

File: product/log-service/handler/stream.go (hunk 4 above)
Comment (smjt-h) [reply product/log-service/handler/stream.go:135]: done

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/stream/stream.go [hunk 5]
```diff
@@ -17,6 +17,8 @@ import (
 // with the Streamer.
 var ErrNotFound = errors.New("stream: not found")
 
+const LogStreamPrefix = "log-service-redis:"
```
Comment (vistaarjuneja) [inline product/log-service/stream/stream.go:20]: Nit: this file should be redis agonistic. So we could just name "Prefix" and use "log-service" as the prefix name

**Review by vistaarjuneja — DISMISSED** [review]

**Comment by smjt-h** [comment]
trigger uts

**Review by vistaarjuneja — APPROVED** [review]

**Comment by smjt-h** [comment]
trigger uts

**Pull request merged**